modules/problem_1/main.cpp

```diff
@@ -0,0 +1,64 @@
+//
+// Created by alyosharomanov on 10/28/2024.
+//
+
+#include "main.h"
+
+#include <algorithm>
+#include <stack>
+#include <vector>
+
+int Solution::LargestRectangleArea(const std::vector<int>& heights) {
+    std::stack<int> indices;
+    int maxArea = 0;
+    const size_t size = heights.size();
+
+    for (size_t i = 0; i <= size; i++) {
+        // do while the current height is shorter than the top of the stack.
+        while (!indices.empty() && (i == size || heights[indices.top()] >= heights[i])) {
+            // height of the rectangle.
+            int height = heights[indices.top()];
+            indices.pop();
+
+            // width of the rectangle.
+            int width;
```

Comment:
avoid code like this, where a variable is not initialized at declaration. 
One option is to initialize it, `int foo{}`. 
Another, often better, option, is to do something like 
   `int foo = cond ? x : y;